llama_index/llms/llama_cpp.py

```diff
@@ -88,8 +93,10 @@ def __init__(
                 "`https://github.com/abetlen/llama-cpp-python`"
             )
 
-        model_kwargs = model_kwargs or {}
-        model_kwargs.update({"n_ctx": context_window, "verbose": verbose})
+        model_kwargs = {
+            **{"n_ctx": context_window, "verbose": verbose},
+            **(model_kwargs or {}),  # Override defaults via model_kwargs
```

Comment:
What if `n_ctx` is in the model_kwargs as well, I think this will error out? (The previous method with update I think handles that case)

Reply:
If `n_ctx` is in `model_kwargs`, it will win out.  With Python `dict` unions done like this, the later-one wins:

```python
In [1]: {**{"hi": 5}, **{"hi": 6}}
Out[1]: {'hi': 6}
```

So please check me if you want, but it should be fine

Reply:
it works, thanks for confirming!